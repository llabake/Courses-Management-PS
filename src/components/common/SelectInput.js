import React from 'react';
import {
  string,
  func,
  arrayOf,
  object,
} from 'prop-types';

const SelectInput = ({
  name,
  label,
  onChange,
  defaultOption,
  value,
  options,
  error,
}) => (
  <div className="form-group">
    <label htmlFor={name}>{label}</label>
    {error && <div className="alert alert-danger">{error}</div>}
    <div className="field">
      <select
        name={name}
        value={value}
        className="form-control"
        onChange={onChange}
      >
        <option value="">{defaultOption}</option>
        {
          options.map(option => (
            <option
              value={option.value}
              key={option.value}
            >
              {option.text}
            </option>
          ))
        }
      </select>
    </div>
  </div>
);

export default SelectInput;
