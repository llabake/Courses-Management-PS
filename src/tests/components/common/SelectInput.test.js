import React from 'react';
import { mount } from 'enzyme';
import SelectInput from "../../../components/common/SelectInput";

describe('<SelectInput />', () => {
  it('field component renders correctly', () => {
    const props = {
      label: '',
      placeholder: '',
      name: '',
      error: [],
      value: '',
      options: [{ value: '', text: '' }],
      defaultOption: '',
      onChange: jest.fn(),
    };
    const wrapper = mount(<SelectInput {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('Select input field component show error when they occur', () => {
    const props = {
      label: 'Author',
      placeholder: '',
      name: 'authorId',
      defaultOption: 'Select Author',
      error: ['Author name must contain at least 4 characters'],
      value: '',
      options: [{ value: '', text: '' }],
      onChange: jest.fn(),
    };
    const wrapper = mount(<SelectInput {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('.alert-danger').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
