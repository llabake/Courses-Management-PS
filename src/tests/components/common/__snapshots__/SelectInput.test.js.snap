// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> Select input field component show error when they occur 1`] = `
<SelectInput
  defaultOption="Select Author"
  error={
    Array [
      "Author name must contain at least 4 characters",
    ]
  }
  label="Author"
  name="authorId"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "text": "",
        "value": "",
      },
    ]
  }
  placeholder=""
  value=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="alert alert-danger"
    >
      Author name must contain at least 4 characters
    </div>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="authorId"
        onChange={[MockFunction]}
        value=""
      >
        <option
          value=""
        >
          Select Author
        </option>
        <option
          key=""
          value=""
        />
      </select>
    </div>
  </div>
</SelectInput>
`;

exports[`<SelectInput /> field component renders correctly 1`] = `
<SelectInput
  defaultOption=""
  error={Array []}
  label=""
  name=""
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "text": "",
        "value": "",
      },
    ]
  }
  placeholder=""
  value=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="alert alert-danger"
    />
    <div
      className="field"
    >
      <select
        className="form-control"
        name=""
        onChange={[MockFunction]}
        value=""
      >
        <option
          value=""
        />
        <option
          key=""
          value=""
        />
      </select>
    </div>
  </div>
</SelectInput>
`;
